chalice/deploy/deployer.py

```diff
@@ -733,6 +762,31 @@ def handle_restapi(self, config, resource):
         resource.swagger_doc = swagger_doc
 
 
+class LambdaEventSourcePolicyInjector(BaseDeployStep):
+    def __init__(self):
+        # type: () -> None
+        self._policy_injected = False
+
+    def handle_sqseventsource(self, config, resource):
+        # type: (Config, models.SQSEventSource) -> None
+        # The sqs integration works by polling for
+        # available records so the lambda function needs
+        # permission to call sqs.
+        role = resource.lambda_function.role
+        if (not self._policy_injected and
```

Comment:
Hmm this is a pretty long if statement. I wonder if it is possible to simplify this?

Reply:
I originally had it simplified, but mypy's flow control doesn't seem to like it and I'd prefer not to explicitly cast.

```python
    def handle_sqseventsource(self, config, resource):
        # type: (Config, models.SQSEventSource) -> None
        # The sqs integration works by polling for
        # available records so the lambda function needs
        # permission to call sqs.
        role = resource.lambda_function.role
        if self._policy_injected:
            # We only need to inject the policy one.
            return
        is_correct_type = (
            isinstance(role, models.ManagedIAMRole) and
            isinstance(role.policy, models.AutoGenIAMPolicy) and
            not isinstance(role.policy.document, models.Placeholder)
        )
        if is_correct_type:
(line 785)  self._inject_trigger_policy(role.policy.document,
                                        SQS_EVENT_SOURCE_POLICY.copy())
            self._policy_injected = True
```

```
$ make typecheck
mypy --py2 --ignore-missing-imports --follow-imports=skip -p chalice --disallow-untyped-defs --strict-optional --warn-no-return
chalice/deploy/deployer.py:785: error: "IAMRole" has no attribute "policy"
make: *** [typecheck] Error 1
```

Reply:
Yeah that does look better. Unfortunate about the mypy problem. I'm fine keeping it as is.